x/stakeibc/keeper/icacallbacks_delegate.go

```diff
@@ -31,12 +31,17 @@ func (k Keeper) UnmarshalDelegateCallbackArgs(ctx sdk.Context, delegateCallback
 	return &unmarshalledDelegateCallback, nil
 }
 
-func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack *channeltypes.Acknowledgement_Result, args []byte) error {
+func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, txMsgData *sdk.TxMsgData, args []byte) error {
 	k.Logger(ctx).Info("DelegateCallback executing", "packet", packet)
 
-	if ack == nil {
-		// transaction on the host chain failed
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "ack is nil, packet %v", packet)
+	if txMsgData == nil {
```

Comment:
I actually really like this approach of differentiating timeout vs ack error!
I think we might wanna log them as `Error` instead of `Info` though

Reply:
good call! will update now